src/helpers/fastPR-url-rules.ts

```diff
@@ -0,0 +1,32 @@
+import { digitalTextbooksUrlParser } from './digitalTextbooks-url-parser';
+import { openDiggerUrlParser } from './openDigger-url-parser';
+import { OSS101TextbooksUrlParser } from './OSS101Textbooks-url-parser';
+
+type UrlRule = {
+  domains: string[];
+  ruleFunction: (url: string) => { filePath: string; repoName: string; branch: string; platform: string } | null;
+};
+const urlRules: UrlRule[] = [
```

Comment:
It seems that these files shouldn't be here; `src/helpers` should contain files that are globally used, and those related to specific functionalities should be in their respective feature folders, or perhaps there could be subdirectories within `helpers`. 

Additionally, if the parsers are all bundled and released within the HyperCRX extension, does it mean that adding a new config requires releasing a new version? An ideal approach would be for all configurations to be in a remote JavaScript file that exports an object containing configurations for all relevant websites. The FastPR feature could then read this file to load configuration data, instead of embedding the configurations within the package.

Reply:
Yes, the next step is to prepare to obtain the configuration file through a remote JavaScript file. I haven't figured out where to put this file for maintenance yet.

Reply:
> Additionally, if the parsers are all bundled and released within the HyperCRX extension, does it mean that adding a new config requires releasing a new version? An ideal approach would be for all configurations to be in a remote JavaScript file that exports an object containing configurations for all relevant websites. The FastPR feature could then read this file to load configuration data, instead of embedding the configurations within the package.

Yes, the next step is to prepare to obtain the configuration file through a remote JavaScript file. I haven't figured out where to put this file for maintenance yet.

Reply:
It could be a file in HyperCRX repo and we can add a workflow to publish the file to OSS after master brach push, this makes the file under git control and fetched from remote.

Reply:
> It could be a file in HyperCRX repo and we can add a workflow to publish the file to OSS after master brach push, this makes the file under git control and fetched from remote.

I tried to dynamically load JS files in OSS, but for our plugin project, there is a security policy issue that reports' `contentScript. bundle. js: 123584 Unconcause (in promise) EvalError: Rejused to evaluate a string as JavaScript because 'unsafe evaluation' is not an allowed source of script in the following Content Security Policy directive: 'script src' elf '' wasm unsafe evaluation '' 'inline specification rules' http://localhost: *  http://127.0.0.1: *... ".`.

Reply:
I have encountered this situation before with some libraries that require dynamic loading. I have not yet explored a good solution for injecting scripts into pages using plugins within our plugin framework. Can we deploy this script to the service and request it?

Reply:
Is that possible to change the `content_security_policy` config of the extension to load scripts from remote, like add `oss.x-lab.info` into it and allow it to load scripts from the domain?

Reply:
I have tried, but it doesn't seem to work. And in this case, every time another document website is added, it will have to be changed again

Reply:
Did you load it as a JS file or a text file? Can you load the file as a pure text file and try to use `eval` to parse it?

Reply:
I am currently loading it as a text file and then using eval or Function to parse it.

Reply:
Thank you for Frank's help. The configuration file was changed to be obtained remotely in OSS using the following method. https://developer.chrome.com/docs/extensions/how-to/security/sandboxing-eval